### PR TITLE
Added Category model

### PR DESCRIPTION
**Pull request by sovisou**

### CODE REVIEW

File: src/main/java/com/onlinebookstore/controller/BookController.java [hunk 1]
```diff
@@ -73,4 +73,11 @@ public BookDto updateBook(@PathVariable Long id, @RequestBody @Valid
     public List<BookDto> search(BookSearchParameters searchParameters) {
         return bookService.search(searchParameters);
     }
+
+    @PutMapping("/{id}/books")
+    @Operation(summary = "Get books by category id",
+            description = "Get a list of books by its category identifier")
+    public List getBooksByCategoryId(Long id) {
```
Comment (Elena-Bruyako) [inline src/main/java/com/onlinebookstore/controller/BookController.java]:
```suggestion
    public List<BookDtoWithoutCategoryIds> getBooksByCategoryId(@PathVariable Long id) {
```

File: src/main/java/com/onlinebookstore/controller/CategoryController.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    @ResponseStatus(HttpStatus.CREATED)
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
+    public CategoryDto createCategory(@RequestBody @Valid CreateCategoryRequestDto requestDto) {
+        return categoryService.save(requestDto);
+    }
+
+    @GetMapping
+    @Operation(summary = "Get all categories",
+            description = "Get a list of all available categories")
+    public List getAll(Pageable pageable) {
```
Comment (Elena-Bruyako) [inline src/main/java/com/onlinebookstore/controller/CategoryController.java]: It's better to return Page

File: src/main/java/com/onlinebookstore/model/Book.java [hunk 3]
```diff
@@ -34,4 +42,13 @@ public class Book {
     private String coverImage;
     @Column(nullable = false)
     private boolean isDeleted = false;
+    @ManyToMany(fetch = FetchType.LAZY)
```
Comment (Elena-Bruyako) [inline src/main/java/com/onlinebookstore/model/Book.java]:
```suggestion
    @ManyToMany
```

File: src/main/java/com/onlinebookstore/mapper/CategoryMapper.java [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+import com.onlinebookstore.config.MapperConfig;
+import com.onlinebookstore.dto.category.CategoryDto;
+import com.onlinebookstore.dto.category.CreateCategoryRequestDto;
+import com.onlinebookstore.model.Category;
+import org.mapstruct.Mapper;
+import org.mapstruct.MappingTarget;
+
+@Mapper(config = MapperConfig.class)
+public interface CategoryMapper {
+
```
Comment (Elena-Bruyako) [inline src/main/java/com/onlinebookstore/mapper/CategoryMapper.java]: remove empty line

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

File: src/main/java/com/onlinebookstore/service/impl/BookServiceImpl.java [hunk 5]
```diff
@@ -64,4 +68,12 @@ public List<BookDto> search(BookSearchParameters parameters) {
                 .map(bookMapper::toDto)
                 .toList();
     }
+
+    @Override
+    public List<BookDtoWithoutCategoryIds> getBooksByCategoryId(Long categoryId) {
+        List<Book> books = bookRepository.findAllByCategories_Id(categoryId);
+        return books.stream()
+                .map(bookMapper::toDtoWithoutCategories)
+                .collect(Collectors.toList());
```
Comment (Elena-Bruyako) [inline src/main/java/com/onlinebookstore/service/impl/BookServiceImpl.java]:
```suggestion
                .toList();
```

File: src/main/java/com/onlinebookstore/service/CategoryService.java [hunk 6]
```diff
@@ -0,0 +1,19 @@
+package com.onlinebookstore.service;
+
+import com.onlinebookstore.dto.category.CategoryDto;
+import com.onlinebookstore.dto.category.CreateCategoryRequestDto;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+
+public interface CategoryService {
+
```
Comment (Elena-Bruyako) [inline src/main/java/com/onlinebookstore/service/CategoryService.java]:
```suggestion
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
just two comments

**Review by sarakhmen — APPROVED** [review]

**Pull request merged**